src/snapred/backend/data/LocalDataService.py

```diff
@@ -210,9 +210,10 @@ def _readPVFile(self, runId: str):
         return f
 
     @ExceptionHandler(StateValidationException)
-    def _generateStateId(self, runId: str) -> Tuple[Any, Any]:
+    def _generateStateId(self, runId: str) -> Tuple[str, str]:
         if runId in self.stateIdCache:
-            return self.stateIdCache[runId]
+            SHA = self.stateIdCache[runId]
+            return SHA.hex, SHA.decodedKey
```

Comment:
This return value isn't necessary if you use an `if/else`

Reply:
That's correct.  However, I'm following the "coding standard" of using _early_ return rather than deep if / else chaining.  It's interesting, as when I learned software development, we were _strongly_ encouraged to have only one return statement per function.

Reply:
**And, possibly more importantly, I'm trying to modify the original code as little as possible.**

Reply:
For me it depends, early escape cases like caching and validation I think are acceptable.